src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/RejectedFilesHandler.java

```diff
@@ -93,27 +95,29 @@ private void handleEnvelope(
         }
     }
 
-    private byte[] download(BlobClient blobClient) throws IOException {
-        try (var outputStream = new ByteArrayOutputStream()) {
-            blobClient.getBlockBlobClient().download(outputStream);
-            return outputStream.toByteArray();
-        }
-    }
-
-    private void upload(BlobClient blobClient, byte[] blobContent, String loggingContext) {
+    private void upload(BlobClient targetBlob, BlobClient sourceBlob, String loggingContext) {
         logger.info(
-            "File uploading to url: {}, isSnapshot: {}",
-            blobClient.getBlockBlobClient().getBlobUrl(),
-            blobClient.getBlockBlobClient().isSnapshot()
+            "File uploading from url: {}, to url {} ",
+            sourceBlob.getBlockBlobClient().getBlobUrl(),
+            targetBlob.getBlockBlobClient().getBlobUrl()
         );
 
-        blobClient
-            .getBlockBlobClient()
-            .upload(
-                new ByteArrayInputStream(blobContent),
-                blobContent.length,
-                true
-            );
+        targetBlob
+            .getBlockBlobClient().copyFromUrl(sourceBlob.getBlobUrl() + "?" + createSas(sourceBlob));
+
         logger.info("File successfully uploaded to rejected container. " + loggingContext);
     }
+
+    private String createSas(BlobClient sourceBlob) {
+        OffsetDateTime expiryTime =
+            OffsetDateTime.of(LocalDateTime.now().plus(5, ChronoUnit.MINUTES), ZoneOffset.UTC);
```

Comment:
may there be time differences between aks and blob ?? , 5 minutes is  enough?

Reply:
both should be updated so should be fine